feature-libs/cart/base/components/clear-cart/clear-cart-dialog/clear-cart.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@angular/core';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+import { filter, switchMap, take, tap } from 'rxjs/operators';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { LaunchDialogService } from '@spartacus/storefront';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ClearCartService {
```

Comment:
Given that this service seems dedicated to the ClearCartDialogComponent, it would be more consistent to call it ClearCartDialogComponentService (we have this pattern with other components I think)